packages/data-table-server/src/dataTableService/utils/dataTableParamsToYupSchema.ts

```diff
@@ -0,0 +1,53 @@
+/**
```

Comment:
Probs better as a chat but... Biao and I were talking about yup today. With the new @tupaia/types package, we want to generate as much as possible to avoid staleness etc. 

He's having a look at generating json-schema from the typescript type definitions, which would then be passed into a json-schema validator like https://github.com/ajv-validator/ajv. This would mean we would get validation for free across all packages. Let's have a look at moving over to this new way?